src/Essentials/src/Flashlight/Flashlight.ios.cs

```diff
@@ -6,6 +6,17 @@ namespace Microsoft.Maui.Devices
 {
 	class FlashlightImplementation : IFlashlight
 	{
+		/// <summary>
+		/// Checks if the flashlight is available and can be turned on or off.
+		/// </summary>
+		/// <returns><see langword="true"/> when the flashlight is available, or <see langword="false"/> when not</returns>
+		public Task<bool> CheckIsSupported()
+		{
+			var captureDevice = AVCaptureDevice.GetDefaultDevice(AVMediaTypes.Video);
```

Comment:
This is me asking questions. We are getting the default video device, but we are not saying the position.

There [is an overload of this method ](https://learn.microsoft.com/en-us/dotnet/api/avfoundation.avcapturedevice.getdefaultdevice?view=xamarin-ios-sdk-12#avfoundation-avcapturedevice-getdefaultdevice(avfoundation-avcapturedevicetype-system-string-avfoundation-avcapturedeviceposition)) that specifies the position of the device (front or back). Can we use the front camera flash as a flashlight? If we cannot, we should specify the position directly to ensure we are using the correct device.

Reply:
The code was copied from `Toggle()`, below. I'll check what the default parameter is. I think most devices on iOS only have a back side flashlight, so I guess that would be the better side to check.

Reply:
I reviewed the two `AVCaptureDevice.GetDefaultDevice()` methods. I would keep te same call as in the `Toggle()` call, below in the same source file, when switching on/off the flashlight. If you want to change what `GetDefaultDevice()` call to use, in order to give the caller the possibility to choose which flashlight to switch, feel free to do so. I think this PR is not the proper context for that, though. I also don't think I saw any iOS devices yet that had a front camera flashlight (but once had an iPad mini that had no back flashlight).